src/main/java/com/endava/internship/collections/StudentMap.java

```diff
@@ -1,72 +1,396 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Consumer;
 
 public class StudentMap implements Map<Student, Integer> {
+
+    private final int DEFAULT_CAPACITY = 1 << 4;
```

Comment:
also make it static, the SNAKE_CASE that we use applyis only to constants. it is not a constant without the static